src/components/MenuModal/MenuModal.tsx

```diff
@@ -0,0 +1,65 @@
+import { ReactNode, useCallback, useRef, useState } from "react";
+import {
+  ContentWrapper,
+  Background,
+  HeadWrapper,
+  MenuImage,
+  ModalWrapper,
+  OcaImage,
+  RetractImage,
+} from "./MenuModal.styles";
+import OcaLogo from "@/../public/logo-oca-simple.svg";
+import RetractIcon from "@/../public/retract-icon.svg";
+import MenuIcon from "@/../public/modal-icon.svg";
+
+const MenuModal = ({
+  children,
+  hasBackground = true,
+  startRetracted = true,
+  ...props
+}: {
+  children?: ReactNode;
+  hasBackground?: boolean;
+  startRetracted?: boolean;
+}) => {
+  const [retracted, setRetracted] = useState<boolean>(startRetracted);
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = useCallback((event: MouseEvent) => {
+    if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
+      setRetracted(true);
+      document.removeEventListener("click", handleClickOutside);
+    }
+  }, []);
+
+  const switchRetract = useCallback(() => {
+    if (retracted) {
+      document.addEventListener("click", handleClickOutside);
+    } else {
+      document.removeEventListener("click", handleClickOutside);
+    }
+
+    console.log("oi", retracted)
+    setRetracted(previous => !previous);
+  }, [handleClickOutside, retracted]);
+
+  return (
+    <div {...props}>
+      <a onClick={switchRetract}>
+        <MenuImage src={MenuIcon} alt="" />
+      </a>
```

Comment:
Why not put the onclick directly on the image?
```suggestion
      <MenuImage onClick={switchRetract} src={MenuIcon} alt="" />
```

Reply:
Because the Next's Image component doesn't accept the onClick prop, we have to wrap it with another tag

Reply:
I changed it from a tag to a div tag to be more semantically correct